File: registration-login-spring-boot-security-thymeleaf-rdbms/src/main/java/com/knf/dev/service/UserService.java
package com.knf.dev.service;

import java.util.List;

import org.springframework.security.core.userdetails.UserDetailsService;

import com.knf.dev.dto.UserRegistrationDto;
import com.knf.dev.model.User;

public interface UserService extends UserDetailsService {
	User save(UserRegistrationDto registrationDto);
	List<User> getAll();
}

File: kotlin-spring-main/kotlin-spring-thymeleaf-login-signup/src/main/kotlin/com/knf/dev/demo/kotlinspringthymeleafloginsignup/service/UserService.kt
package com.knf.dev.demo.kotlinspringthymeleafloginsignup.service

import org.springframework.security.core.userdetails.UserDetailsService
import com.knf.dev.demo.kotlinspringthymeleafloginsignup.dto.UserRegistrationDto
import com.knf.dev.demo.kotlinspringthymeleafloginsignup.model.User

interface UserService : UserDetailsService {
    fun save(registrationDto: UserRegistrationDto?): User?
    val all: List<User?>?
}